diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0fbd5ce..e8f7e40 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,7 +33,7 @@ endif()
 SET (ORIGINAL_LIB_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
 
 SET(BOOST_COMPONENTS)
-LIST(APPEND BOOST_COMPONENTS thread date_time system filesystem program_options signals serialization chrono unit_test_framework context locale iostreams)
+LIST(APPEND BOOST_COMPONENTS thread date_time system filesystem program_options serialization chrono unit_test_framework context iostreams)
 
 IF( WIN32 )
   MESSAGE(STATUS "Configuring fc to build on Win32")
diff --git a/include/fc/io/sstream.hpp b/include/fc/io/sstream.hpp
index 9010bfe..fe9f0d7 100644
--- a/include/fc/io/sstream.hpp
+++ b/include/fc/io/sstream.hpp
@@ -27,7 +27,7 @@ namespace fc {
 
     private:
       class impl;
-      fwd<impl,368> my;
+      fwd<impl,512> my;
   };
 
 }
